src/statement.ts

```diff
@@ -0,0 +1,240 @@
+   * @returns A bound Statement.
+   */
+  bind<T = S>(...values: ValuesType[]): Statement<T> {
+    const parameters = getParameters(...values);
+    return new Statement(this.config, this.sql, parameters);
+  }
+
+  /**
+   * Resolve a bound statement to a SQL string.
+   * @returns A valid SQL string.
+   */
+  toString(): string {
+    try {
+      return bindValues(this.sql, this.parameters);
+    } catch (cause) {
+      throw new Error("BIND_ERROR", { cause });
+    }
+  }
+
+  toObject(): { sql: string; parameters?: Parameters } {
+    return {
+      sql: this.sql,
+      parameters: this.parameters,
+    };
+  }
+
+  async #parseAndExtract(): Promise<ExtractedStatement> {
+    const statementWithBindings = this.toString();
+    const { type, statements, tables } = await normalize(statementWithBindings);
+    // Stick with original if a create statement, otherwise, use the parsed version
+    // This is because the parser injects keywords that are not spec compliant
+    // See https://github.com/tablelandnetwork/go-sqlparser/issues/41
+    const sql =
+      type === "create" ? statementWithBindings : statements.join(";");
+    return { type, sql, tables };
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async all<T = S, K extends keyof T = keyof T>(
+    colName?: undefined,
+    opts?: Signal
+  ): Promise<Result<T>>;
+  async all<T = S, K extends keyof T = keyof T>(
+    colName: K,
+    opts?: Signal
+  ): Promise<Result<T[K]>>;
+  async all<T = S, K extends keyof T = keyof T>(
+    colName?: K,
+    opts: Signal = {}
+  ): Promise<Result<T | T[K]>> {
+    try {
+      const start = performance.now();
+      const { sql, type, tables } = await this.#parseAndExtract();
+      switch (type) {
+        case "read": {
+          const config = await extractReadonly(this.config, {
+            type,
+            tables,
+          });
+          const results = await queryAll<T>(config, sql, opts);
+          if (colName != null) {
+            return wrapResult(
+              extractColumn(results, colName),
+              performance.now() - start
+            );
+          }
+          return wrapResult(results, performance.now() - start);
+        }
+        default: {
+          let receipt = await exec(this.config, { type, sql, tables });
+          if (this.config.autoWait ?? false) {
+            const waited = await receipt.wait();
+            receipt = { ...receipt, ...waited };
+          }
+
+          return wrapResult(receipt, performance.now() - start);
+        }
+      }
+    } catch (cause) {
+      if (
+        cause instanceof Error &&
+        cause.message.includes("column not found")
+      ) {
+        throw new Error("COLUMN_NOTFOUND", { cause });
+      }
+      throw new Error("ALL_ERROR", { cause });
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async first<T = S, K extends keyof T = keyof T>(): Promise<T>;
```

Comment:
Not much special here, basically the same as all.

Reply:
I've been testing this, and it probably should have been obvious at first ;) but
```
const res = await db.prepare(`SELECT * FROM ${tableName};`).first();
console.log(res);
// { col: "val1" }
```
and
```
const res = await db.prepare(`SELECT * FROM ${tableName};`).all();
console.log(res);
/*
{
  meta: { duration: 17.52889347076416 },
  success: true,
  results: [ { col: 'val1'}, { col: 'val2' } ]
}
*/
```

Basically first returns a single row as an object and all returns an array with each row as an object.  Dobule checking I understand this correctly?

Reply:
Yes your interpretation is correct. It is really just a shortcut for puling out the first result. Honestly kinda seems like a silly method. In theory, if we were using real prepared statements, then this method makes sense, because you'd just retrieve the first found record.